Zend/zend_compile.h

```diff
@@ -479,13 +480,15 @@ typedef struct _zend_class_constant {
 typedef struct _zend_internal_arg_info {
 	const char *name;
 	zend_type type;
+	zend_type_node *type_tree;
```

Comment:
This increases size of one of the core data structure, and therefore the common memory usage.

I think ``zend_type_node`` and ``zend_type`` should be somehow combined together.